cpp/open3d/visualization/rendering/filament/FilamentCamera.cpp

```diff
@@ -253,9 +242,14 @@ Camera::Transform FilamentCamera::GetViewMatrix() const {
     return FilamentToCameraTransform(ftransform);
 }
 
-Camera::Transform FilamentCamera::GetProjectionMatrix() const {
-    auto ftransform = camera_->getProjectionMatrix();  // mat4 (not mat4f)
-    return FilamentToCameraTransform(ftransform);
+Camera::ProjectionMatrix FilamentCamera::GetProjectionMatrix() const {
+    auto ft = camera_->getProjectionMatrix();  // mat4 (not mat4f)
+    Camera::ProjectionMatrix::MatrixType proj;
+    proj << float(ft(0, 0)), float(ft(0, 1)), float(ft(0, 2)), float(ft(0, 3)),
```

Comment:
maybe with double(...) or static_cast<double>(...) ?

Reply:
If you look into the guts, Camera::ProjectionMatrix::MatrixType ends up being a float matrix, not a double. Also, I personally hate static_cast (and think that the usually stated reason for using it, "easier to search for" is lame, because I have never once done or heard of anyone searching for casts, while on the negative side it uses underscores which are hard to type, masquerade as spaces visually which has a one pixel line as differentiator between a real space which ends a token, and takes up a lot of space in lines which are usually already pretty cramped for space), and in this case it makes things much less readable, since you can fit the 4 columns of the matrix in one line.